torch/_higher_order_ops/cond.py

```diff
@@ -168,12 +170,15 @@ def _validate_input(pred, true_fn, false_fn, operands):
     def _cond_op_wrapper(*args, **kwargs):
         return cond_op(*args, **kwargs)
 
-    with _set_compilation_env():
-        with torch._dynamo.utils.disable_cache_limit():
-            with _temp_remove_pre_dispatch_torch_function_mode():
-                return torch.compile(_cond_op_wrapper, backend="eager", fullgraph=True)(
-                    pred, true_fn, false_fn, operands
-                )
+    with _set_compilation_env(), torch._dynamo.utils.disable_cache_limit(), _temp_remove_pre_dispatch_torch_function_mode():
+        with _temp_remove_metadata_torch_function_mode() as metadata_mode:
```

Comment:
Do you mind also changing associative_scan's implementation? https://github.com/pytorch/pytorch/blob/main/torch/_higher_order_ops/associative_scan.py#L128-L132.

Reply:
Yup I'll update

Reply:
oops, I forgot flex attention: https://github.com/pytorch/pytorch/blob/main/torch/nn/attention/flex_attention.py#L1036-L1040

Reply:
associative_scan doesn't use eager?